File: backend/src/main/java/com/cduestc/openatom/approval/entity/Approval.java
package com.cduestc.openatom.approval.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("approvals")
public class Approval {
    @TableId(type = IdType.ASSIGN_ID)
    private Long id;
    
    private String type; // join_club, activity, project, points
    private Long applicantId;
    private String title;
    private String description;
    private String status; // pending, approved, rejected
    private Long reviewerId;
    private String reviewComment;
    private LocalDateTime reviewedAt;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/types/approval.ts
export interface Approval {
  id: number
  type: string // join_club, activity, project, points
  applicantId: number
  applicantName?: string
  title: string
  description: string
  status: string // pending, approved, rejected
  reviewerId: number
  reviewerName?: string
  reviewComment: string
  reviewedAt: string
  createdAt: string
  updatedAt: string
}

export interface ApprovalQuery {
  page?: number
  size?: number
  type?: string
  status?: string
  applicantId?: number
}

export interface PageResult<T> {
  records: T[]
  total: number
  page: number
  size: number
}
